backend/app/synthesize_polygons/synthesize_polygon.py

```diff
@@ -103,29 +123,56 @@ def generate_note_with_amplitude(frequency, duration, amplitude):
     :return: numpy array which represents the note
     """
     time_steps = np.linspace(0, duration, int(duration * WAV_SAMPLE_RATE), False)
-    note = np.sin(frequency * time_steps * 2 * np.pi) * amplitude
+    note = np.sin(frequency * time_steps * 2 * np.pi) * amplitude * amplitude_decay(frequency)
     return note
 
 
+def generate_time_stamps(duration_list, note_delay, sides_as_duration=False):
+    """
+    Generates a list indicating when the sound associated with each line of the polygon should
+    start and end in seconds.
+    :param duration_list: a list of durations of each side
+    :param note_delay: delay between each note in seconds
+    :param sides_as_duration: whether to use side lengths to determine duration. if False, then use
+    side lengths to determine amplitude.
+    :return: list of tuples signifying the time in which each line starts and ends in seconds
+    """
+
+    time_stamps = []
+    start = 0
+    for duration in duration_list:
+        end = start + duration
+        time_stamps.append((start, end))
+        if sides_as_duration:
+            start += duration
+        else:
+            start += note_delay
+
+    return time_stamps
+
+
 # pylint: disable=too-many-arguments
 # pylint: disable=too-many-locals
-# TODO: Fill in for and use restrict_octave
-# TODO: Fill in for and use sides_as_duration along with sides_as_duration function
-def synthesize_polygon(points, note_length=1, note_delay=0, restrict_octave=False,
-                       sides_as_duration=False, base_frequency=220):
+def synthesize_polygon(points, note_length=1, note_delay=1, restrict_frequency=False,
```

Comment:
Cleanup TODO: this function is now getting stuffed enough that I would consider breaking apart into helper functions. The code smell for me is that we have a lot of conditional branches such that it's pretty tricky to read a straight codepath.

What I'd probably do is something like refactor into `synthesize_polygon_sides_as_duration` and `synthesize_polygon_sides_as_frequency` and then make the common code into helper functions of both.